src/emmetHelper.ts

```diff
@@ -383,14 +383,29 @@ function getFilters(text: string, pos: number): { pos: number, filter: string }
 /**
  * Extracts abbreviation from the given position in the given document
  */
-export function extractAbbreviation(document: TextDocument, position: Position, lookAhead: boolean = true): { abbreviation: string, abbreviationRange: Range, filter: string } {
+export function extractAbbreviation(document: TextDocument, position: Position, extractOptions: boolean | { lookAhead?: boolean, syntax?: string }): { abbreviation: string, abbreviationRange: Range, filter: string } {
 	const currentLine = getCurrentLine(document, position);
 	const currentLineTillPosition = currentLine.substr(0, position.character);
 	const { pos, filter } = getFilters(currentLineTillPosition, position.character);
 	const lengthOccupiedByFilter = filter ? filter.length + 1 : 0;
 
 	try {
-		const result = extract(currentLine, pos, lookAhead);
+		let options: boolean | { lookAhead: boolean, syntax: string };
+		if (typeof extractOptions === 'boolean') {
+			options = extractOptions;
+		} else {
+			let syntax: string;
+			let lookAhead: boolean;
+			if (extractOptions.syntax && isStyleSheet(extractOptions.syntax)) {
```

Comment:
check on `extractOptions.syntax` not needed here. `isStyleSheet` returns `false` if null/undefined/emptystring is passed